### ISSUE TITLE
[Feature request]: fake /home directory with linkable object

### ISSUE THREAD

**Issue by morfyum** [flatpak/flatpak#5043]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

Hello! 
Im just a casual user. I have an issue with flatpak apps, which infect my $HOME directory.
For example: I install vscodium and install extension in vscodium.
After remove adware/malware extension, the sneaky files stan on my home directory.
Remove extension or whole app doesnt remove this.

So my question:
Can I separate my whole home directory from flatpak apps?

My vision:
```
# mount/link fake folders/files to my home direcotry, and when app is removed, my home is cleand from flatpak_app_1 contents
# flatpak_app_1 always use own fake home dir.
# Required folders are mountable to my home.
# for example:
## mount (somehow)  flatpak_app_1/fakehome/folder_to_grant_app_1/   /home/user1/folder_to_grant_app_1/
## `ln (somehow) flatpak_app_1/fakehome/folder_to_grant_app_1/something  /home/user1/folder_to_grant_app_1/something` 

/home/user1/
  - Downloads/
       some-file-by-flatpak_app_1   -> flatpak_app_1/fakehome/Downloads/some-file-by-flatpak_app_1
  - Videos/
  - Pictures/
  - Music/
  - folder_to_grant_app_1/     ->   link-from-flatpak_app_1
      other-file-by-flatpak_app_1.txt

flatpak_app_1/fakehome/
  - Downloads/
      some-file-by-flatpak_app_1
  - Videos/
  - Pictures/
  - Music/
  - folder_to_grant_app_1/
```

I dont know is it possible or complete nonsense.
Could you give me some advice, please?
How can I protect my `$HOME` from aggressive apps?
How can I keep clean my `$HOME`? 

Thanks!

**Comment by rusty-snake** [comment]
> How can I protect my $HOME from aggressive apps?
How can I keep clean my $HOME?

Remove `home` and `host` permission.

For the rest of you setup you can use `persist` and symlink the folder.

For vscodium you can try something like 

```bash
flatpak override --user --nofilesystem=host --persist=foo com.vscodium.codium && ln -s ~/.var/app/com.vscodium.codium/foo ~/foo
```

**Comment by nanonyme** [comment]
Right, so what is being requested here is default flatpak behaviour. Certain apps request access to real home but others have a virtual home instead.

**Comment by morfyum** [comment]
Thanks for answer!

I remove `host`, and the issue maybe gone.
The Flatpak built in `~` is automatically cleand after close, I see well?
And the `foo` linked directory content is persistent, so this is the mountpoint.

This solution is reverse of mine, but acceptable.


Thanks for help!
Possibly can you recommend a safe installation parameters for flatpak apps, please?
I would like to set as alias. After installation I solve manage the permission issues.

**Comment by morfyum** [comment]
> Right, so what is being requested here is default flatpak behaviour. Certain apps request access to real home but others have a virtual home instead.

I understand.

This problem is puzzling to me.
The virtual homes are not persistent. (after restart vscodium, the home content removed)

This solution: 
  - not force virtual home, applications has oppurtunity to acces my home.
  - Home is not persistent, so its stay clean (not as my /home when I not force this)
  - I can insert access point to applications.

My ide:
  - Force virtual home, and keep a list about created home files. ( possibly 2 list: 1 for persistent data and 1 for temporary)
  - Virtual home persistent `or` persistent `and` temporary lists of mountpoints link to my real home.
  - After remove or exit applications, end this relationship
 

My idea is more difficult, and maybe impossible to solve safe, true?

**Comment by nanonyme** [comment]
Virtual homes are partially persisted. Eg Steam app sets --persist=. to persist all of it.
